src/groups.py

```diff
@@ -34,8 +34,12 @@ def expand_nodes(node_list):
     del nodes[-1]
     return nodes
 
-def __nodeattr(file_path='{}genders'.format(config.LEADER), arguments=[], split_char="\n"):
+def __nodeattr(file_path='{}genders'.format(config.LEADER),
+               arguments=[],
+               split_char="\n"):
     try:
         return local['nodeattr']['-f', file_path](arguments).split(split_char)
+    # in the case that the genders file can't be accessed, we're acting as if
+    # there's an empty genders file in that location
     except ProcessExecutionError as e:
```

Comment:
It isn't a good idea to rely on an error from the underling `nodeattr` command as it may change. Instead a check for the missing genders file should be done upfront.

Reply:
Also remove the above comment as it will be obvious what the check is doing

Reply:
The comment was there as more of a statement of policy rather than an explanation of the code but sure, I'll take it out

Reply:
YEah, done here https://github.com/alces-software/adminware/pull/169/commits/79982bef3b7f4c4157ea2cd375bd0a81f2392dfe